sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,76 @@ export class EventHubClient {
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
-      );
-    }
+    let connectionString;
+    let sharedAccessKeyName = "defaultKeyName";
+    let sharedAccessKey = "defaultKeyValue";
+    let eventHubPath: string | undefined;
+    let tokenProvider = {} as TokenProvider;
 
-    if (!entityPath) {
-      throw new Error("Please provide a value for the 'entityPath' parameter.");
-    }
+    hostOrConnectionString = String(hostOrConnectionString);
+    if (typeof eventHubPathOrOptions === "string") {
+      eventHubPath = eventHubPathOrOptions;
+      if (!eventHubPath) {
+        throw new Error("Please provide a value for the 'entityPath' parameter.");
+      }
+      if (!tokenProviderOrCredentials || typeof {}) {
+        throw new Error("Please provide either a token provider or a valid credentials object.");
+      }
 
-    if (!tokenProviderOrCredentials) {
-      throw new Error("Please provide either a token provider or a valid credentials object.");
-    }
+      if (
+        tokenProviderOrCredentials instanceof ApplicationTokenCredentials ||
+        tokenProviderOrCredentials instanceof UserTokenCredentials ||
+        tokenProviderOrCredentials instanceof DeviceTokenCredentials ||
+        tokenProviderOrCredentials instanceof MSITokenCredentials
+      ) {
+        tokenProvider = new AadTokenProvider(tokenProviderOrCredentials);
+      } else {
+        tokenProvider = tokenProviderOrCredentials;
+        if (tokenProvider instanceof SasTokenProvider) {
+          sharedAccessKeyName = tokenProvider.keyName;
+          sharedAccessKey = tokenProvider.key;
+        }
+      }
 
-    let sharedAccessKeyName = "defaultKeyName";
-    let sharedAccessKey = "defaultKeyValue";
-    let tokenProvider: TokenProvider;
-    if (
-      tokenProviderOrCredentials instanceof ApplicationTokenCredentials ||
-      tokenProviderOrCredentials instanceof UserTokenCredentials ||
-      tokenProviderOrCredentials instanceof DeviceTokenCredentials ||
-      tokenProviderOrCredentials instanceof MSITokenCredentials
-    ) {
-      tokenProvider = new AadTokenProvider(tokenProviderOrCredentials);
+      let host = hostOrConnectionString;
+      if (!host) {
+        throw new Error(
+          "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
+        );
+      }
+      if (!host.endsWith("/")) host += "/";
+      connectionString = `Endpoint=sb://${host};SharedAccessKeyName=${sharedAccessKeyName};SharedAccessKey=${sharedAccessKey};`;
```

Comment:
If we add the event hub path to the connection string here, then we can skip passing the event hub path to `EventHubConnectionConfig.create()` and make it into a constant inside the `if` block